src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -725,6 +728,32 @@ public <T> T load(InputStream in, Class<T> clazz) throws IOException {
         return obj;
     }
 
+    /**
+     * Loads an object from a resource, if it exists.
+     *
+     * @param resource The resource to read the object from.
+     * @param clazz The class of the expected object.
+     * @throws IOException if something went wrong.
+     * @throws FileNotFoundException if the file does not exist.
+     */
+    public <T> T load(Resource resource, Class<T> clazz) throws IOException {
+        Resource.Type type = resource.getType();
+
+        if (type == Resource.Type.RESOURCE) {
+            try (InputStream in = resource.in()) {
+                // use a BufferedInputStream (if it not already is).
+                InputStream bin =
+                        in instanceof BufferedInputStream ? in : new BufferedInputStream(in, 512);
```

Comment:
the instanceof is not really necessary until no one has the idea to implement it into Resource.in() itself. (See discussion on [GSIP 155](https://github.com/geoserver/geoserver/pull/2060#pullrequestreview-18679072))